pipeline_dp/aggregate_params.py

```diff
@@ -113,45 +113,83 @@ def metrics_str(self) -> str:
             return f"custom combiners={[c.metrics_names() for c in self.custom_combiners]}"
         return f"metrics={[m.value for m in self.metrics]}"
 
+    @property
+    def bounds_per_value_are_set(self) -> bool:
+        return self.min_value is not None and self.max_value is not None
+
+    @property
+    def bounds_per_partition_are_set(self) -> bool:
+        return self.min_sum_per_partition is not None and self.max_sum_per_partition is not None
+
     def __post_init__(self):
         if self.low is not None:
             raise ValueError(
                 "AggregateParams: please use min_value instead of low")
         if self.high is not None:
             raise ValueError(
                 "AggregateParams: please use max_value instead of high")
-        if self.metrics:
-            needs_min_max_value = Metrics.SUM in self.metrics \
-                                  or Metrics.MEAN in self.metrics \
-                                  or Metrics.VARIANCE in self.metrics
-            if needs_min_max_value and (self.min_value is None or
-                                        self.max_value is None):
-                raise ValueError(
-                    "AggregateParams: min_value and max_value must be set")
-            if needs_min_max_value and (_not_a_proper_number(self.min_value) or
-                                        _not_a_proper_number(self.max_value)):
-                raise ValueError(
-                    "AggregateParams: min_value and max_value must be both "
-                    "finite numbers")
-            if needs_min_max_value and self.max_value < self.min_value:
-                raise ValueError(
-                    "AggregateParams: max_value must be equal to or greater than"
-                    " min_value")
-            if Metrics.VECTOR_SUM in self.metrics and \
-            (Metrics.SUM in self.metrics or \
-            Metrics.MEAN in self.metrics or \
-            Metrics.VARIANCE in self.metrics):
+        if (self.min_value is None) != (self.max_value is None):
```

Comment:
It seems we have logic in multiple places to check this (and `min/max_sum_per_partition`), can we consolidate them somehow?